### PR TITLE
chores: update the logic of calculating bundle profit and block reward

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: miner/bidder.go [hunk 1]
```diff
@@ -190,24 +206,24 @@ func (b *bidder) bid(work *environment) {
...
 // isBestWork returns the work is better than the current best work
-func (b *bidder) isBestWork(work *environment) bool {
-	if work.blockReward == nil {
+func (b *Bidder) isBestWork(work *environment) bool {
+	if work.bundleProfit == nil {
 		return false
 	}
 
-	return b.getBestWork(work.header.Number.Int64()).blockReward.Cmp(work.blockReward) < 0
+	return b.getBestWork(work.header.Number.Int64()).bundleProfit.Cmp(work.bundleProfit) < 0
```
Comment (irrun) [inline miner/bidder.go:214]: en? why only consider bundleprofit

File: miner/bidder.go [hunk 2]
```diff
@@ -157,16 +177,12 @@ func (b *bidder) bid(work *environment) {
 			ParentHash:  parent.Hash(),
 			GasUsed:     work.header.GasUsed,
 			GasFee:      work.blockReward.Uint64(),
-			// TODO(roshan) decide builderFee according to realtime traffic and validator commission
-			BuilderFee: big.NewInt(int64(float64(work.bundleProfit.Uint64() * 5 / 100))),
-			Txs:        txs,
-			Timestamp:  time.Now().Unix(),
+			Txs:         txs,
+			Timestamp:   time.Now().Unix(),
```
Comment (irrun) [inline miner/bidder.go:181]: TODO should be keeped

File: miner/worker.go [hunk 3]
```diff
@@ -716,14 +710,20 @@ func (w *worker) commitTransaction(env *environment, tx *txpool.Transaction, rec
 		env.gasPool.SetGas(gp)
 		return nil, err
 	}
+
 	env.txs = append(env.txs, tx.Tx)
 	env.receipts = append(env.receipts, receipt)
 
+	gasUsed := new(big.Int).SetUint64(receipt.GasUsed)
+	env.bundleProfit.Add(env.bundleProfit, gasUsed.Mul(gasUsed, tx.Tx.GasPrice()))
+	env.blockReward.Add(env.blockReward, gasUsed.Mul(gasUsed, tx.Tx.GasPrice()))
```
Comment (irrun) [inline miner/worker.go:719]: is common tx goes into here?

**Review by irrun — COMMENTED** [review]
define bundleProfit and blockReward

File: miner/bidder.go (hunk 2 above)
Comment (Pythonberg1997) [reply miner/bidder.go:181]: Done

**Review by irrun — APPROVED** [review]

**Pull request merged**